healthcheck.go

```diff
@@ -0,0 +1,53 @@
+package redis
+
+import (
+	"context"
+	"errors"
+
+	health "github.com/ONSdigital/dp-healthcheck/healthcheck"
+
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+const MsgHealthy = "redis is healthy"
+
+var (
+	ErrorFailedConnection = errors.New("couldn't connect to redis")
+)
+
+// Checker executes all healthchecks and then updates the health state
+func (cli *Client) Checker(ctx context.Context, state *health.CheckState) error {
+	if state == nil {
+		state = &health.CheckState{}
+	}
+
+	statusCode, err := cli.healthcheck(ctx)
+	if err != nil {
+		if updateErr := state.Update(health.StatusCritical, err.Error(), statusCode); updateErr != nil {
+			log.Warn(ctx, "unable to update health state", log.FormatErrors([]error{updateErr}))
+		}
+
+		return nil
+	}
+
+	if updateErr := state.Update(health.StatusOK, MsgHealthy, statusCode); updateErr != nil {
+		log.Warn(ctx, "unable to update health state", log.FormatErrors([]error{updateErr}))
+	}
+
+	log.Info(ctx, "checkstate", log.Data{
```

Comment:
Something that's bugged me for some time is our libraries doing their own logging. Apps should log, libraries should return results which are then logged if wanted by the thing that uses them. Ie. the health checker calling this check should log the response of the check. This pattern is replicated across most of our helper libraries though so maybe not something to address here, but just thought I should mention it. It should be possible to use a library's functions in an app that doesn't want to log it without unwanted side-effects.